src/cairoWriter/writers/memberAccessWriter.ts

```diff
@@ -1,8 +1,12 @@
-import { ASTWriter, MemberAccess, SrcDesc } from 'solc-typed-ast';
+import { ASTWriter, MemberAccess, SrcDesc, TypeNameType } from 'solc-typed-ast';
 import { CairoASTNodeWriter } from '../base';
 
 export class MemberAccessWriter extends CairoASTNodeWriter {
   writeInner(node: MemberAccess, writer: ASTWriter): SrcDesc {
-    return [`${writer.write(node.vExpression)}.${node.memberName}`];
+    if (this.ast.inference.typeOf(node.vExpression) instanceof TypeNameType) {
+      return [`${writer.write(node.vExpression)}::${node.memberName}`];
```

Comment:
Could you provide a sample line of solidity code and the generated cairo code for that case?

Reply:
Sure.
```
pragma solidity ^0.8.0;

library c {
  function CALLXY() external pure returns (int24) {
    return 1;
  }
}

contract WARP {
  function f() external pure returns (int24) {
    return c.CALLXY();
  }
}
```

```
    #[view]
    fn f_26121ff0()-> (__warp_0 : felt){
        let __warp_pse_0 = WARP::CALLXY_1f1273a5();
        return __warp_pse_0;
    }
```